src/DynamoCoreWpf/Views/Core/NodeView.xaml

```diff
@@ -262,32 +291,83 @@
             </Border.Visibility>
         </Border>
 
+        <StackPanel Orientation="Horizontal"
+                    Grid.Row="1"
+                    Grid.ColumnSpan="3"
+                    Panel.ZIndex="12"
+                    HorizontalAlignment="Center"
+                    Margin="3 0 3 0">
+            <TextBlock Name="NameBlock"
+                       VerticalAlignment="Center"
+                       HorizontalAlignment="Stretch"
+                       Grid.ColumnSpan="3"
+                       Text="{Binding Name, UpdateSourceTrigger=PropertyChanged}"
+                       Padding="5"
+                       FontSize="13"
+                       FontWeight="SemiBold"
+                       FontFamily="{StaticResource OpenSansSemibold}"
+                       TextAlignment="Center"
+                       IsHitTestVisible="False"
+                       Background="{x:Null}"
+                       Style="{StaticResource SZoomFadeText}">
+                <TextBlock.Foreground>
+                    <Binding Path="State"
+                             UpdateSourceTrigger="PropertyChanged"
+                             Mode="OneWay"
+                             ConverterParameter="HeaderForeground"
+                             Converter="{StaticResource StateToColorConverter}">
+                    </Binding>
+                </TextBlock.Foreground>
+            </TextBlock>
+
+            <Border CornerRadius="3,3,3,3"
+                    Height="12"
+                    Width="Auto"
+                    Background="{StaticResource renamedLabelBackground}"
+                    Margin="3 0 3 0">
+                <Border.Style>
+                    <Style TargetType="{x:Type Border}">
+                        <Setter Property="Visibility"
+                                Value="Collapsed" />
+                        <Style.Triggers>
+                            <DataTrigger Binding="{Binding IsRenamed}"
+                                         Value="True">
+                                <Setter Property="Visibility"
+                                        Value="Visible" />
+                            </DataTrigger>
+                        </Style.Triggers>
+                    </Style>
+                </Border.Style>
+                <TextBlock Text="Renamed"
+                           HorizontalAlignment="Center"
+                           VerticalAlignment="Center"
+                           FontStyle="Italic"
```

Comment:
Unsure if this needs to be italic, the other `tags` we're adding to preview bubble aren't iirc, so I'd keep them consistent

Reply:
all tags are in italic